### PR TITLE
Add envoy config to expose prom endpoint for scraping

### PR DESCRIPTION
**Pull request by douglas-reid**
This PR is meant to address #8415. It provides a mechanism for removing the dependency on the statsd-to-prometheus process while maintaining prometheus-based access to important per-proxy configuration and cache behavior metrics.

It exposes the `/stats/prometheus` endpoint for access and adds custom tags to the `stats_config` for envoy (enabling actual consumption of
istio-configured cluster-based stats that include the '|' character and working around issues with the default set of tags).

It also adds the following metadata annotations to the `kube-injection` (and gateway/istio component proxies):

```yaml
prometheus.io/path: /stats/prometheus
prometheus.io/port: "15090"
prometheus.io/scrape: "true"
```

Finally, this PR sets up a blacklist in the prometheus configmap (editable by operators) to reduce the burden of storing massive amounts of metrics data.

Using this PR, it is now possible to issue the following query:

```
sum(envoy_cluster_manager_cds_version) by (pod_name)
```

And receive results like:

```
{pod_name="details-v1-56cb65b78f-9rc6l"}	507876366460273000
{pod_name="productpage-v1-6955cc5fdf-lqsw4"}	507876366460273000
```

This PR also removes the deployment of the `statsd-to-prometheus` process and disables the `envoyStatsd` bit of configuration by default (behavior is preserved for legacy and other cases).

As part of this PR, the Pilot dashboard was updated slightly to use the new metrics.

### CODE REVIEW

File: tools/deb/envoy_bootstrap_v2.json [hunk 1]
```diff
@@ -47,6 +47,19 @@
   },
   "static_resources": {
     "clusters": [
+    {
+      "name": "prometheus_stats",
+      "type": "LOGICAL_DNS",
```
Comment (kyessenov) [inline tools/deb/envoy_bootstrap_v2.json]: `STATIC` is better, no need to introduce DNS dependency.

**Comment by douglas-reid** [comment]
Reviewers: Please take a look. If this looks like a reasonable approach, I will update the necessary unit/e2e tests and add configuration options (where appropriate).

I'd like feedback on (a) what should be configurable (prometheus port? envoy scraping altogether?) and (b) if the initial metrics blacklist looks reasonable.

**Comment by douglas-reid** [comment]
Reviewers: All of the tests are now passing. Please take a look. I believe this has the configurability requested (keyed off `prometheusPort` variable in helm `values.yaml`) and meets all of the criteria for usage.

**Pull request merged**